infra/docker-compose/docker-compose.dev.yml

```diff
@@ -0,0 +1,70 @@
+version: "3.7"
+
+services:
+  core:
+    image: maven:3.6-openjdk-11
+    volumes:
+      - ${HOME}/.m2:/root/.m2:delegated
+      - ../../.:/code:cached
```

Comment:
Not sure about using `cached` here. According to the [docs](https://docs.docker.com/docker-for-mac/osxfs-caching/#cached):
> For directories mounted with cached, the host’s view of the file system is authoritative; writes performed by containers are immediately visible to the host, but there may be a delay before writes performed on the host are visible within containers.

This can be confusing if I make a change in the code in the host but when I run with `mvn -pl core spring-boot run` the change is not reflected.

Reply:
indeed, it's something that I thought about but I don't know if it's an issue.  All changes should be reflected on startup but changes from the host while the container is running may be delayed.  I'm not sure the performance is worth it but I'm not sure when it will be a problem.  I only did the bare minimum to get my dev setup.  I'm planning on further changes as I haven't even completed a full build, debug, change, build, cycle yet.  I opened this since I struggled so much to get everything running that I thought it would be worth it and it's low risk.

Reply:
https://docs.docker.com/docker-for-mac/osxfs-caching/ gives an overview of the issue with fs performance on OSX.  I could also see an argument for reversing the settings (.m2 cached and /code delegated).  The reason I did it this way was I expected most code changes would be on the host and m2 changes during build on the container.  Given your other comment about using `docker-compose exec` to keep the container running, I could see reversing these settings.  I'm not really sure and haven't run this setup before, so open to any advice.

Reply:
With more reading, I think this is the right approach because the performance seems pretty bad.  The way I interpret it is, `delegated` for stuff you can lose, `cached` for faster reads, `consistent` when you can't deal with the delay.  I don't know what the delay is but I prefer this unless there is an objection.  I will resolve this but feel free to unresolve if you think this is not the right approach.